src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs

```diff
@@ -48,14 +48,15 @@ public static partial class JsonSerializer
                         break;
                 }
 
-                if (flushThreshold >= 0 && writer.BytesPending > flushThreshold)
+                if (finishedSerializing && writer.CurrentDepth == 0)
                 {
-                    return false;
+                    continueWriting = false;
```

Comment:
Should this just be a `break,` and then we can get rid of `continueWriting` all together and the loop can be `while(true)`?

Reply:
yep it's better

Reply:
done